packages/apputils/src/dialog.ts

```diff
@@ -43,18 +43,29 @@ export function showErrorMessage(
   title: string,
   error: any,
   buttons: ReadonlyArray<Dialog.IButton> = [
-    Dialog.okButton({ label: 'DISMISS' })
+    Dialog.okButton({ label: 'Dismiss' })
   ]
 ): Promise<void> {
   console.warn('Showing error:', error);
 
-  return showDialog({
-    title: title,
-    body: error.message || title,
-    buttons: buttons
-  }).then(() => {
-    /* no-op */
-  });
+  // Cache promises to prevent multiple copies of identical dialogs showing
+  // to the user.
+  let body = error.message || title;
+  let key = title + '----' + body;
+  let promise = Private.errorMessagePromiseCache.get(key);
+  if (promise) {
+    return promise;
+  } else {
+    let dialogPromise = showDialog({
+      title: title,
+      body: body,
+      buttons: buttons
+    }).then(() => {
+      Private.errorMessagePromiseCache.delete(key);
+    });
```

Comment:
`catch` also?

Reply:
Makes sense. Can you open an issue or PR?